api/app/controllers/spree/api/checkouts_controller.rb

```diff
@@ -34,8 +34,13 @@ def update
           end
 
           return if after_update_attributes
-          state_callback(:after) if @order.next
-          respond_with(@order, default_template: 'spree/api/orders/show')
+
+          if @order.completed? || @order.next
+            state_callback(:after)
+            respond_with(@order, default_template: 'spree/api/orders/show')
+          else
+            respond_with(@order, default_template: 'spree/api/orders/could_not_transition', status: 422)
```

Comment:
Line is too long. [104/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.